lib/playlistGenerator/ConcatSource.js

```diff
@@ -439,7 +439,12 @@ ConcatSource.prototype.checkDiscontinuity = function (fileInfo,track) {
                 }
                 firstDTS = that.inner.refPTS + dtsDistance(track.firstEncoderDTS, that.inner.refEncoderDTS,track.wrapEncoderDTS);
             }
-
+            let diagDrift = {
+                deltaPts : track.firstDTS - that.inner.refPTS,
+                deltaClock : track.firstEncoderDTS - that.inner.refEncoderDTS
+            };
```

Comment:
Rather save it as propery than sending event
Also please add the refpts and refencoderdts